templates/workloads/partials/cf/envvars.yml

```
@@ -28,11 +28,9 @@ Environment:
   Value: !Sub
     - '{{jsonQueueURIs .Subscribe.Topics}}'
     - mainURL: !Ref EventsQueue
-    {{- range $topic := .Subscribe.Topics}}
-    {{- if $topic.Queue}}
-      {{$topic.Service}}-{{$topic.Name}}URL: !Ref {{logicalIDSafe $topic.Service}}{{logicalIDSafe $topic.Name}}EventsQueue
-    {{- end}}
-    {{- end}}
+    {{- range $topic := .Subscribe.Topics}}{{- if $topic.Queue}}{{- if $topic.Service}}{{- if $topic.Name}}
```

We can make use of `and` function: https://pkg.go.dev/text/template#hdr-Functions
```suggestion
    {{- range $topic := .Subscribe.Topics}}{{- if and $topic.Queue $topic.Service $topic.Name}}
```

---

does this work with more than two arguments?

---

yess it does